src/cryptography/hazmat/backends/openssl/x509.py

```diff
@@ -107,6 +108,26 @@ def _build_general_name(backend, gn):
         return x509.DirectoryName(
             _build_x509_name(backend, gn.d.directoryName)
         )
+    elif gn.type == backend._lib.GEN_EMAIL:
```

Comment:
This function is getting rather large and every branch in the main if immediately returns. Maybe it should be multiple functions?